app/controllers/api/base_controller.rb

```diff
@@ -19,4 +19,22 @@ def enqueue_web_hook_jobs(version)
       )
     end
   end
+
+  def verify_with_otp
+    otp = request.headers["HTTP_OTP"]
+    return if @api_key.user.mfa_api_authorized?(otp)
+    prompt_text = otp.present? ? t(:otp_incorrect) : t(:otp_missing)
+    render plain: prompt_text, status: :unauthorized
+  end
+
+  def find_api_key
+    params_key = request.headers["Authorization"] || params.permit(:api_key).fetch(:api_key, "")
+    hashed_key = Digest::SHA256.hexdigest(params_key)
```

Comment:
It is recommended that we use Bcrypt for hashing passwords as SHA256 is fast hashing implementation. I don't think we need Bcrypt for API keys as they are random.